packages/app-desktop/gui/ConfigScreen/ConfigScreen.tsx

```diff
@@ -570,21 +576,31 @@ class ConfigScreenComponent extends React.Component<any, any> {
 				const onTextChange = (event: any) => {
 					updateSettingValue(key, event.target.value);
 				};
-
 				return (
 					<div key={key} style={rowStyle}>
 						<div style={labelStyle}>
 							<label>{md.label()}</label>
 						</div>
-						<input
-							type={inputType}
-							style={inputStyle}
-							value={this.state.settings[key]}
-							onChange={(event: any) => {
-								onTextChange(event);
-							}}
-							spellCheck={false}
-						/>
+						{
+							md.subType === SettingItemSubType.FontFamily ?
+								<FontSearch
+									_key={key}
```

Comment:
Why is `_key` needed?

Reply:
The main purpose of passing the key is to be able to update the settings value (`updateSettingValue`), as it takes `key` as the first argument. But if you are asking about the naming and why didn't I use just `key` because key is a reserved word in react components and when using it, I get this warning ` ``key`` is not a prop. Trying to access it will result in ``undefined`` being returned. If you need to access the same value within the child component, you should pass it as a different prop. `

Reply:
The normal pattern for a React controller component is to have a "value" and "onChange" (or similar) event. If you do that you won't have to pass around unnecessary properties and can create more reusable components.

So in this case, add an `onChange` event to your component which would pass to the parent the selected font. Then here just do `onChange={fontFamily => updateSettingValue(key, fontFamily)}`

I know there's a lot of code in this project and you can't know everything, but it often helps to see how things are done in the code near where you want to add something. For example you can see that all components here follow this value/onChange pattern. Maybe also read about controlled components and why there generally preferred.